chat/src/components/general/ChatInput.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+import { ReactComponent as RightArrowIcon } from "../../assets/right-arrow.svg";
+
+type Props = {
+  onSubmit: (message: string) => void;
+  isDisabled: boolean;
+};
+export function ChatInput({
+  onSubmit,
+  isDisabled,
+  ...props
+}: Props): React.ReactElement {
+  const [message, setMessage] = useState("");
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    textareaRef.current?.focus();
+  }, []);
+
+  return (
+    <Wrapper {...props}>
+      <RightArrowContainer
+        onClick={() => {
+          onSubmit(message);
+          setMessage("");
+        }}
+      >
+        <RightArrowPadding>
+          <RightArrowIcon />
+        </RightArrowPadding>
+      </RightArrowContainer>
+      <Textarea
+        ref={textareaRef}
+        autoFocus
+        placeholder="Type here what you need, or select some code"
+        value={message}
+        onChange={(e) => {
+          setMessage(e.target.value);
+        }}
+        onKeyPress={(e) => {
+          if (
+            e.key === "Enter" &&
+            e.shiftKey === false &&
+            message.trim().length > 0 &&
+            !isDisabled
+          ) {
+            e.preventDefault();
+            onSubmit(message.trim());
+            setMessage("");
+          }
+        }}
+      />
+    </Wrapper>
+  );
+}
+
+const Wrapper = styled.div`
+  position: relative;
+  border-top: 1px solid #433f47;
+`;
+
+const RightArrowContainer = styled.div`
+  position: absolute;
+  right: 8px;
+  top: 9px;
```

Comment:
Padding maybe? not sure

Reply:
I think it's ok. it should be painted on the textarea